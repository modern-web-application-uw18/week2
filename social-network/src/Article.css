a {
    text-decoration: none;
}
/* Article CSS */
.Article {
    box-sizing: border-box;
    width: 45%;
    background-color: white;
    border-color: gray;
    border-style: solid;
    border-width: 1pt;
    flex-direction: row;
    margin: 5px;
}
  
  .Article-image {
    box-sizing: border-box;
    width: 200px;
    height: 300px;
    object-fit: contain;
    overflow: hidden; 
    align: left;
    float: left;
  }

  .Article-body-container {
      box-sizing: border-box;
      width: 50%;
      height: 50%;
      padding: 15px;
      float: left;
    }

  .Article-header {
    color: black;
    font-weight: bold;
    font-size: 16px;
    text-align: left;
    word-wrap: normal;
  }
  
   .Article-description {
    color: gray;
    font-size: 13px;
    text-align: left;
}

  .Author-container {
    height: 50%;
    padding: 15px;
    float: bottom;   
    }

    .Author-image {
        width: auto;
        height: auto;
        max-width: 50px;
        max-height: 50px;
        border-radius: 100%;
        overflow: hidden;    
        margin: 10px;
    }

    .Author-name {
        align-items: center;
        font-size: 12px;
        color: gray;
        margin: 3px;
    }