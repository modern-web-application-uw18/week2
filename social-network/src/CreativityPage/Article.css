
.ForYouSection {
    display: flex;
    flex-wrap: wrap;
    border-top: 1px solid lightgrey;
}
.MissedSection {
    display: flex;
    flex-wrap: wrap;
    border-top: 1px solid lightgrey;
}

.flex-container-box-two-per-row {
    display: flex;
    /*flex-wrap: nowrap;*/
    flex-direction: row;
    border: 1px solid lightgrey;
    margin: 10px;
    width: 45%;
}
.flex-container-box-three-per-row {
    display: flex;
    flex-wrap: wrap;
    /*flex-wrap: nowrap;*/
    flex-direction: row;
    border: 1px solid lightgrey;
    margin: 10px;
    width: 30%;
}

.flex-container-box-generic {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    margin: 10px;
}

/*  Within a box, this is all the stuff right of the image*/
.flex-container-contents {
    display: flex;
    /*flex-wrap: wrap;*/
    flex-direction: column;
    padding-left: 10px;
    max-width: 300px;
}

/*  Within a container-contents, this is author image, name, date and readtime data*/
.flex-container-metadata {
    display: flex;
    /*flex-wrap: wrap;*/
    flex-direction: row;
    padding-bottom: 8px;
    font-size: 14px;
}

/*  Within flex-container-metadata, this is everything to the right of author pic*/
.flex-container-md-column {
    display: flex;
    flex-direction: column;
    padding-left: 10px;
    color: rgba(0,0,0,.54);
}

.flex-container-md-row {
    display: flex;
    /*flex-wrap: wrap;*/
    flex-direction: row;
    align-items: center;
    padding-bottom: 10px;
}

/* Within container-metadata, this is the name, date, readtime */
.flex-container-meta-text {
    display: flex;
    /*flex-wrap: wrap;*/
    flex-direction: column;
}

.flex-container-space-filler {
    flex-grow: 1; /*grow to fill space to push what comes after to bottom of box*/
}
.ArticleMinutesToRead {
    position: relative;
}
.Circle {
	border-radius: 50%;
	width: 3px;
    height: 3px;
    background-color: lightgrey;
    margin: 5px;
}