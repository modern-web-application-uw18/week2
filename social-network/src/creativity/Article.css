* {
    box-sizing: border-box;
}

a:link, a:visited, a:hover, a:active {
    text-decoration: none;
    color: gray;
}

.Article-container {
    display: flex;
    flex-wrap: nowrap;
    height: 220px;
    width: 430px;
    border: lightgray;
    border-width: 1px;
    border-style: solid;
    margin: 8px;
}

.Article-image-container {
    width: 180px;
    height: 217px;
}

.Article-image-container img {
    width: 180px;
    height: 217px;
    overflow: hidden;
}

.Article-main-column {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    height: 216px;
}

.Article-main-column-top {
    height: 168px;
    width: 240px;
    padding: 10px;
}

.Author {
    display: flex;
    flex-wrap: nowrap;
    color: gray;
    font-weight: 500;
    height: 60px;
    width: 284px;
}
