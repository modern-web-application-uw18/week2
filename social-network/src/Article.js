import React, { Component } from 'react';
import './Article.css';
import PropTypes from 'prop-types';

class Article extends Component {
    render() {
        const article = this.props.article;
        return (
            <div className="Article" key>
                <div className="Article-image">
                    <img src={article.image} alt={article["title"]} />
                </div>
                <div className="Article-body-container">
                    <a href={article.link}>
                        <header className="Article-header">{article.title}</header>
                    </a>
                    <div className="Article-description">
                        <p>{article.description}</p>
                    </div>
                    <div className="Author-container">
                        <div className="Author-image">
                            <img src={article.author["image"]} alt={article.author["name"]} />
                        </div>
                        <div className="Author-name">
                            <p>{article.author["name"]}</p>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

Article.propTypes = {
    article: PropTypes.shape({
        image: PropTypes.string,
        title: PropTypes.string,
        link: PropTypes.string,
        description: PropTypes.string,
        author: PropTypes.shape({
            image: PropTypes.string,
            name: PropTypes.string
        })
      })
};

export default Article;
