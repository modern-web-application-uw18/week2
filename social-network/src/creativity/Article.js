import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Description from './Description';
import AudioPreview from './AudioPreview';
import Author from './Author';
import './Article.css';

class Article extends Component {

    render() {

        const {
            title,
            description,
            image,
            link,
            author,
            postedDate,
            minutesToRead,
            hasAudioAvailable,
            memberPreview
        } = this.props.article;

        return (
            <a href={link} target='_blank'>
                <div className="Article-container" >
                    <div className="Article-image-container" >
                        <img src={image} alt="" />
                    </div>
                    <div className="Article-main-column">
                        <div className="Article-main-column-top">
                            <AudioPreview hasAudioAvailable={hasAudioAvailable} memberPreview={memberPreview} />
                            <Description title={title} description={description} />
                        </div>
                        <Author author={author} postdate={postedDate} minutes={minutesToRead} />
                    </div>
                </div >
            </a>
        );
    }
}

Article.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    image: PropTypes.string,
    link: PropTypes.string,
    author: PropTypes.instanceOf(Author),
    postedDate: PropTypes.string,
    minutesToRead: PropTypes.number,
    hasAudioAvailable: PropTypes.bool,
    memberPreview: PropTypes.bool
};

Article.defaultProps = {
    title: 'No title :(',
    description: 'No description :(',
    hasAudioAvailable: false,
    memberPreview: false
};

export default Article;
