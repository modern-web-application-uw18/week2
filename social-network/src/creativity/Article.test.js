import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Article from './Article';
import testArticle from '../json/test-article.json';

describe('Article', () => {

    it('renders without crashing with data', () => {
        const div = document.createElement('div');
        const element = testArticle.map((article, index) => {
            return <Article key={index} article={article} />;
        });
        ReactDOM.render(element, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with title, description, image, etc.', () => {
        const element = testArticle.map((article, index) => {
            return <Article key={index} article={article} />;
        });
        const component = TestRenderer.create(element);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
