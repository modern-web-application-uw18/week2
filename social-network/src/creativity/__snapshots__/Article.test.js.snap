// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should render with title, description, image, etc. 1`] = `
<a
  href="https://artplusmarketing.com/the-7-habits-of-highly-creative-people-6cfe6471af31"
  target="_blank"
>
  <div
    className="Article-container"
  >
    <div
      className="Article-image-container"
    >
      <img
        alt=""
        src="https://cdn-images-1.medium.com/max/800/1*PIC-U670COEqmGwt4cvXAA.jpeg"
      />
    </div>
    <div
      className="Article-main-column"
    >
      <div
        className="Article-main-column-top"
      >
        <div
          className="Description"
        >
          <div
            className="Article-header"
          >
            The 7 Habits of Highly Creative People
          </div>
          <div
            className="Article-description"
          >
            "What a good artist understands is that nothing comes from nowhere. Nothing is completely original."
          </div>
        </div>
      </div>
      <div
        className="Author"
      >
        <div
          className="Author-image-member-container"
        >
          <img
            alt=""
            src="https://cdn-images-1.medium.com/fit/c/80/80/1*NGHEMBPNbIQmmAkvCjaspA.jpeg"
          />
        </div>
        <div
          className="Author-text-container"
        >
          <div
            className="Author-name"
          >
            Louis Chew
          </div>
          <div
            className="Author-post"
          >
            Apr 04
             Â· 
            8
             min read
          </div>
        </div>
        <div
          className="Article-bookmark-container"
        >
          <img
            alt="Bookmark"
            src="icon-bookmark.svg"
          />
        </div>
      </div>
    </div>
  </div>
</a>
`;
